src/common/interactive_retriever.py

```diff
@@ -0,0 +1,214 @@
+"""
+Retrieves candidates for a given set of queries after embedding them.
+"""
+
+from enum import Enum
+import gc
+import json
+import os
+
+import numpy as np
+import torch
+import torch.distributed as dist
+from torch.utils.data import DataLoader
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+from data.mbeir_dataset import (
+    MBEIRInferenceOnlyDataset,
+    MBEIRInferenceOnlyCollator,
+)
+import dist_utils
+from dist_utils import ContiguousDistributedSampler
+from mbeir_embedder import generate_embeds_and_ids_for_dataset_with_gather
+from utils import build_model_from_config, set_seed
+from data.preprocessing.utils import unhash_did
+
+
+class QueryModality(Enum):
+    TEXT = "text"
+    IMAGE = "image"
+
+
+class InteractiveRetriever:
+    def __init__(self, cand_index_path: str, candidates_path: str, config):
+        # Set up seed for reproducibility
+        seed = config.seed + dist_utils.get_rank()
+        set_seed(seed)
+
+        # MSCOCO's dataset id is hardcoded since the dataset id and query/candidate modalities determine the instruction part of the prompt.
+        # MSCOCO's dataset supports prompt instructions for both image->text and text->image query->candidate modalities.
+        self.dataset_id = 9
```

Comment:
Is the InteractiveRetriever specifically designed for the MSCOCO dataset? I observed that the `self.dataset_id `and `task_id` assignments appear to be hardcoded.